packages/netlify-cms-backend-bitbucket/src/implementation.ts

```diff
@@ -216,7 +219,16 @@ export default class BitbucketBackend implements Implementation {
     if (!isCollab) {
       throw new Error('Your BitBucket user account does not have access to this repo.');
     }
-
+    if (!this.isBranchConfigured) {
+      const defaultBranchName = await getDefaultBranchName({
+        backend: 'bitbucket',
+        repo: this.repo,
+        token: this.token,
+      });
+      if (defaultBranchName) {
+        this.branch = defaultBranchName;
```

Comment:
Indeed, I kept wrestling with the question of when to set `this.branch`. I put it after the check for user having write access because I thought there was no point setting branch when user is unauthenticated or doesn't have write access.

What do you recommend? Setting branch before calling `hasWriteAccess`, or even earlier than `authenticate`?